{
  "repository": {
    "name": "nosan/embedded-cassandra",
    "githubId": 137877866,
    "url": "https://github.com/nosan/embedded-cassandra",
    "isFork": false,
    "original": {
      "name": "",
      "githubId": 0,
      "url": ""
    },
    "isPullRequest": false,
    "pullRequestId": 0
  },
  "builds": {
    "buggyBuild": {
      "id": 401312837,
      "url": "http://travis-ci.org/nosan/embedded-cassandra/builds/401312837",
      "date": "Jul 7, 2018 6:05:15 PM"
    }
  },
  "reproductionBuggyBuild": {
    "reproductionDateBeginning": "Jul 7, 2018 6:05:36 PM",
    "reproductionDateEnd": "Jul 7, 2018 9:54:45 PM",
    "totalDuration": 13746,
    "processDurations": {
      "cloning": {
        "nbSteps": 1,
        "totalDuration": 3,
        "stepNames": [
          "CloneRepository"
        ],
        "stepDurations": [
          3
        ]
      },
      "building": {
        "nbSteps": 2,
        "totalDuration": 87,
        "stepNames": [
          "CheckoutBuggyBuild",
          "BuildProject"
        ],
        "stepDurations": [
          0,
          87
        ]
      },
      "testing": {
        "nbSteps": 1,
        "totalDuration": 49,
        "stepNames": [
          "TestProject"
        ],
        "stepDurations": [
          49
        ]
      },
      "fixing": {
        "nbSteps": 3,
        "totalDuration": 13585,
        "stepNames": [
          "Nopol",
          "NPEFix",
          "AstorJMut"
        ],
        "stepDurations": [
          13509,
          0,
          76
        ]
      }
    }
  },
  "patchDiff": {
    "files": {
      "numberAdded": 0,
      "numberChanged": 0,
      "numberDeleted": 0
    },
    "lines": {
      "numberAdded": 0,
      "numberDeleted": 0
    }
  },
  "projectMetrics": {
    "numberSourceFiles": 30,
    "numberTestFiles": 23,
    "numberLibraries": 31
  },
  "tests": {
    "failingModule": "/root/workspace/nosan/embedded-cassandra/401312837/embedded-cassandra",
    "overallMetrics": {
      "numberRunning": 40,
      "numberPassing": 30,
      "numberFailing": 3,
      "numberErroring": 7,
      "numberSkipping": 0,
      "failures": [
        {
          "failureName": "java.lang.AssertionError",
          "isError": false,
          "occurrences": 3
        },
        {
          "failureName": "java.io.IOException",
          "isError": true,
          "occurrences": 7
        }
      ]
    },
    "failingClasses": [
      {
        "testClass": "com.github.nosan.embedded.cassandra.CassandraTests",
        "numberRunning": 3,
        "numberPassing": 0,
        "numberFailing": 1,
        "numberErroring": 2,
        "numberSkipping": 0
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "numberRunning": 7,
        "numberPassing": 0,
        "numberFailing": 2,
        "numberErroring": 5,
        "numberSkipping": 0
      }
    ],
    "failureDetails": [
      {
        "testClass": "com.github.nosan.embedded.cassandra.CassandraTests",
        "testMethod": "successfullyStarted",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.CassandraTests",
        "testMethod": "impossibleToStartTwice",
        "failureName": "java.lang.AssertionError",
        "detail": "\nExpected: (an instance of java.io.IOException and exception with message a string containing \"Cassandra has already been started\")\n     but: exception with message a string containing \"Cassandra has already been started\" message was \"Could not start a process: \u003cEOF\u003e\"\nStacktrace was: java.io.IOException: Could not start a process: \u003cEOF\u003e\n at com.github.nosan.embedded.cassandra.process.CassandraProcess.onAfterProcessStart(CassandraProcess.java:107)\n at de.flapdoodle.embed.process.runtime.AbstractProcess.\u003cinit\u003e(AbstractProcess.java:116)\n at com.github.nosan.embedded.cassandra.process.CassandraProcess.\u003cinit\u003e(CassandraProcess.java:61)\n at com.github.nosan.embedded.cassandra.process.CassandraExecutable.start(CassandraExecutable.java:49)\n at com.github.nosan.embedded.cassandra.process.CassandraExecutable.start(CassandraExecutable.java:37)\n at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:108)\n at com.github.nosan.embedded.cassandra.Cassandra.start(Cassandra.java:110)\n at com.github.nosan.embedded.cassandra.CassandraTests.start(CassandraTests.java:73)\n at com.github.nosan.embedded.cassandra.CassandraTests.impossibleToStartTwice(CassandraTests.java:59)\n at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n at java.lang.reflect.Method.invoke(Method.java:498)\n at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)\n at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)\n at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)\n at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)\n at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)\n at org.junit.rules.RunRules.evaluate(RunRules.java:20)\n at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)\n at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)\n at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)\n at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)\n at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)\n at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)\n at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)\n at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)\n at org.junit.runners.ParentRunner.run(ParentRunner.java:363)\n at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)\n at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)\n at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)\n at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)\n at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)\n at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)\n at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)\n",
        "isError": false
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.CassandraTests",
        "testMethod": "restart",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "shouldUseRpcTransport",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "restart",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "multiplyInstances",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "invalidConfig",
        "failureName": "java.lang.AssertionError",
        "detail": "\nExpected: (an instance of java.io.IOException and exception with message a string containing \"Missing required directive CommitLogSync\")\n     but: exception with message a string containing \"Missing required directive CommitLogSync\" message was \"Could not start a process: \u003cEOF\u003e\"\nStacktrace was: java.io.IOException: Could not start a process: \u003cEOF\u003e\n at com.github.nosan.embedded.cassandra.process.CassandraProcess.onAfterProcessStart(CassandraProcess.java:107)\n at de.flapdoodle.embed.process.runtime.AbstractProcess.\u003cinit\u003e(AbstractProcess.java:116)\n at com.github.nosan.embedded.cassandra.process.CassandraProcess.\u003cinit\u003e(CassandraProcess.java:61)\n at com.github.nosan.embedded.cassandra.process.CassandraExecutable.start(CassandraExecutable.java:49)\n at com.github.nosan.embedded.cassandra.process.CassandraExecutable.start(CassandraExecutable.java:37)\n at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:108)\n at com.github.nosan.embedded.cassandra.process.CassandraStarterTests.start(CassandraStarterTests.java:128)\n at com.github.nosan.embedded.cassandra.process.CassandraStarterTests.start(CassandraStarterTests.java:119)\n at com.github.nosan.embedded.cassandra.process.CassandraStarterTests.invalidConfig(CassandraStarterTests.java:85)\n at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n at java.lang.reflect.Method.invoke(Method.java:498)\n at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)\n at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)\n at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)\n at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)\n at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)\n at org.junit.rules.RunRules.evaluate(RunRules.java:20)\n at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)\n at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)\n at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)\n at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)\n at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)\n at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)\n at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)\n at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)\n at org.junit.runners.ParentRunner.run(ParentRunner.java:363)\n at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)\n at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)\n at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)\n at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)\n at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)\n at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)\n at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)\n",
        "isError": false
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "useNativeTransport",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "timeout",
        "failureName": "java.lang.AssertionError",
        "detail": "\nExpected: (an instance of java.io.IOException and exception with message a string containing \"Could not start a process. Please increase your startup timeout.\")\n     but: exception with message a string containing \"Could not start a process. Please increase your startup timeout.\" message was \"Could not start a process: \u003cEOF\u003e\"\nStacktrace was: java.io.IOException: Could not start a process: \u003cEOF\u003e\n at com.github.nosan.embedded.cassandra.process.CassandraProcess.onAfterProcessStart(CassandraProcess.java:107)\n at de.flapdoodle.embed.process.runtime.AbstractProcess.\u003cinit\u003e(AbstractProcess.java:116)\n at com.github.nosan.embedded.cassandra.process.CassandraProcess.\u003cinit\u003e(CassandraProcess.java:61)\n at com.github.nosan.embedded.cassandra.process.CassandraExecutable.start(CassandraExecutable.java:49)\n at com.github.nosan.embedded.cassandra.process.CassandraExecutable.start(CassandraExecutable.java:37)\n at de.flapdoodle.embed.process.runtime.Executable.start(Executable.java:108)\n at com.github.nosan.embedded.cassandra.process.CassandraStarterTests.start(CassandraStarterTests.java:128)\n at com.github.nosan.embedded.cassandra.process.CassandraStarterTests.start(CassandraStarterTests.java:119)\n at com.github.nosan.embedded.cassandra.process.CassandraStarterTests.timeout(CassandraStarterTests.java:95)\n at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n at java.lang.reflect.Method.invoke(Method.java:498)\n at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)\n at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)\n at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)\n at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)\n at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)\n at org.junit.rules.RunRules.evaluate(RunRules.java:20)\n at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)\n at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)\n at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)\n at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)\n at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)\n at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)\n at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)\n at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)\n at org.junit.runners.ParentRunner.run(ParentRunner.java:363)\n at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)\n at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)\n at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)\n at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)\n at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)\n at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)\n at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)\n",
        "isError": false
      },
      {
        "testClass": "com.github.nosan.embedded.cassandra.process.CassandraStarterTests",
        "testMethod": "shouldWorkWithDisableTransport",
        "failureName": "java.io.IOException",
        "detail": "Could not start a process: \u003cEOF\u003e",
        "isError": true
      }
    ]
  },
  "commits": {
    "buggyBuild": {
      "repoName": "nosan/embedded-cassandra",
      "branchName": "refactoring-process",
      "sha": "32a9f9458bd9df12572e56d08cf9481fbb029898",
      "url": "http://github.com/nosan/embedded-cassandra/commit/32a9f9458bd9df12572e56d08cf9481fbb029898",
      "date": "Jul 7, 2018 6:01:18 PM"
    }
  },
  "type": "only_fail"
}